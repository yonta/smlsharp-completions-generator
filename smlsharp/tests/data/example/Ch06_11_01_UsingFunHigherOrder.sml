fun powerCurry 0 C= 1
  | powerCurry n C = C * (powerCurry (n - 1) C);

val square = powerCurry 2;

fun Sigma f 0 = 0
  | Sigma f n = f n + Sigma f (n - 1);

Sigma square 3;
Sigma (powerCurry 3) 3;
