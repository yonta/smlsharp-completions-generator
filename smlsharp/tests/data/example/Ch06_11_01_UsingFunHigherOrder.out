val powerCurry = fn : int -> int -> int

val square = fn : int -> int

val Sigma = fn : (int -> int) -> int -> int

val it = 14 : int

val it = 36 : int
